helix-term/src/ui/mod.rs

```diff
@@ -65,9 +65,28 @@ pub fn regex_prompt(
                 }
                 PromptEvent::Validate => match Regex::new(input) {
                     Ok(regex) => {
+                        let (view, doc) = current_ref!(cx.editor);
+
+                        // If the current view is an empty scratch buffer and is not displayed in any other views, do not push to jump.
+                        // Boolean value is determined before the call to `view_mut` because the operation requires a borrow
+                        // of `self.tree`, which is mutably borrowed when `view_mut` is called.
+                        let prevent_empty_scratch_push = !doc.is_modified()
+                            // If the buffer has no path and is not modified, it is an empty scratch buffer.
+                            && doc.path().is_none()
+                            // If the buffer we are changing to is not this buffer
+                            && doc_id != doc.id()
+                            // Ensure the buffer is not displayed in any other splits.
+                            && !cx.editor
+                                .tree
+                                .traverse()
+                                .any(|(_, v)| v.doc == doc.id() && v.id != view.id);
```

Comment:
This a copy paste of a whole bunch of logic. This would be better suited as a helper method.

Reply:
Wait, does this actually work? This is before the `fun` callback is called, so `doc_id` is always equal to `doc.id()`.